### ISSUE TITLE
pulumi up --target says URN could not be found in stack even though pulumi stack -u shows the URN

### ISSUE THREAD

**Issue by markfickett** [pulumi/pulumi#11375]
### What happened?

I tried to update a specific URN, but even though Pulumi shows the URN and the expected stack modification, it also shows an error that the URN isn't found.

### Steps to reproduce

List the URN I want to update:

```
$ pulumi stack -u | grep 'urn:.*dev.arbin_central.host'
    │     URN: urn:pulumi:aws-prod::tech-data-pipeline-aws::aws:ssm/parameter:Parameter::dev.arbin_central.host
```

Attempt to update it:

```
$ pulumi up --target 'urn:pulumi:aws-prod::tech-data-pipeline-aws::aws:ssm/parameter:Parameter::dev.arbin_central.host'
Previewing update (aws-prod):
     Type                  Name                             Plan       Info
     pulumi:pulumi:Stack   tech-data-pipeline-aws-aws-prod             1 error
 -   └─ aws:ssm:Parameter  dev.arbin_central.host           delete

Diagnostics:
  pulumi:pulumi:Stack (tech-data-pipeline-aws-aws-prod):
    error: Target 'urn:pulumi:aws-prod::tech-data-pipeline-aws::aws:ssm/parameter:Parameter::dev.[secret]_central.host' could not be found in the stack. Did you forget to escape $ in your shell?
```

The "Previewing update" block shows what I would expect, deleting one AWS SSM Parameter. However, the diagnostic seems to list the same item as if it could not be found.

### Expected Behavior

I would expect the update to succeed using the URN target.

### Actual Behavior

I get an error and can't update.

### Output of `pulumi about`

```
$ pulumi about
CLI
Version      3.38.0
Go Version   go1.19
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.7.2
awsx        1.0.0-beta.11
docker      3.4.1
eks         0.42.5
honeycomb   0.0.11
kubernetes  3.21.4
python      unknown

Host
OS       ubuntu
Version  20.04
Arch     x86_64

This project is written in python: executable='/path/to/.../python-3.9.5/bin/python3' version='3.9.5
'

Current Stack: aws-prod

TYPE                                                        URN
...
aws:ssm/parameter:Parameter                                 urn:pulumi:aws-prod::tech-data-pipeline-aws::aws:ssm/parameter:Parameter::dev.arbin_central.host
```

### Additional context

I'm using a slightly older version of Pulumi because I am blocked on updating by the protocol buffer v4 dependency (I use other projects still stuck on v3).

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by thomas11** [comment]
Thanks for the report, @markfickett , I'll be looking into it. The `[secret]` in the not-found url is suspicious. Does the value `arbin` come from a config value that's marked as a secret?

**Comment by aq17** [comment]
Thanks for picking this up @thomas11 !

**Comment by markfickett** [comment]
Thanks for checking ths out @thomas11 ! The `arbin` value is not coming from a config (the `arbin_central` is all one string, not assembled from components). But, there is a secret config value where the actual value is `arbin` elsewhere in the project. Could Pulumi be scanning for instances of any secret's value and swapping it out?

**Comment by thomas11** [comment]
We do indeed have such a secret scrubbing logic, but it's supposed to run only on output, not on URN's before they are used internally. I'll validate this.

I assume you can find the correct URN in your project's stack.json file?

**Comment by thomas11** [comment]
I wrote a small repro but wasn't able to reproduce the issue. I do see the redacted output, but the actual operation succeeds. That leads me to believe the `[secret]` angle is a red herring and there is an issue with your stack.

Can you try `pulumi refresh` to sync Pulumi's state with the AWS state?

**Comment by markfickett** [comment]
I had initially needed to use `--target` because someone else had un-updated Pulumi changes; we got that resolved, so I ran a full `pulumi up` and that was able to process this URN. But unfortunately that means I can't repro any more.

I did take a look at my `stack.json` and the URNs look fine, no evidence of `[secret]`s.

If it comes up again I'll revisit the ticket and try a `refresh`. In the mean time I think it's OK to close this from my perspective.

Thanks for your help!
